File: 2023-2024/sourcecode/mod1/lab1/4.2ParcelableSerializableBasics_Java/app/src/main/java/com/example/a42_parcelableserializablebasics_java/SecondActivity.java
package com.example.a42_parcelableserializablebasics_java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;

import com.example.a42_parcelableserializablebasics_java.databinding.ActivitySecondBinding;

public class SecondActivity extends AppCompatActivity {

    private ActivitySecondBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySecondBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        Properties properties;
        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            properties = getIntent().getParcelableExtra(MainActivity.EXTRA_KEY, Properties.class);
        } else {
            properties = getIntent().getParcelableExtra(MainActivity.EXTRA_KEY);
        }

        SerializableProperties serializableProperties;
        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            serializableProperties = getIntent().getSerializableExtra(MainActivity.EXTRA_SERIALIZABLE, SerializableProperties.class);
        } else {
            serializableProperties = (SerializableProperties) getIntent().getSerializableExtra(MainActivity.EXTRA_SERIALIZABLE);
        }

        String result = "";

        if(properties != null && serializableProperties != null)
            result = properties.getA() + properties.getB() + " " + properties.getC() + "---" + serializableProperties.getA() + "\n" +
                    serializableProperties.getB() + " " + serializableProperties.getC();

        binding.textView.setText(result);
    }
}

File: 2023-2024/sourcecode/mod1/lab1/41ParcelableSerializableBasicsKotlin/app/src/main/java/com/example/a41_parcelableserializablebasics_kotlin/SecondActivity.kt
package com.example.a41_parcelableserializablebasics_kotlin

import android.os.Build
import android.os.Build.VERSION.SDK_INT
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.a41_parcelableserializablebasics_kotlin.databinding.ActivitySecondBinding

class SecondActivity : AppCompatActivity() {

    private val binding: ActivitySecondBinding by lazy { ActivitySecondBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        val prop =
            if (SDK_INT >= Build.VERSION_CODES.TIRAMISU)
                intent.getParcelableExtra(EXTRA_KEY, Properties::class.java) // min api 33
            else
                intent.getParcelable(EXTRA_KEY) // api 1 - 32

        val serializableProperties =
            if (SDK_INT >= Build.VERSION_CODES.TIRAMISU)
                intent.getSerializableExtra(EXTRA_SERIALIZABLE, SerializableProperties::class.java) // min api 33
            else
                intent.getSerializable(EXTRA_SERIALIZABLE) // api 1 - 32


        val parcelizeProperties =
            if (SDK_INT >= Build.VERSION_CODES.TIRAMISU)
                intent.getParcelableExtra(EXTRA_PARCELIZE, ParcelizeProperties::class.java) // min api 33
            else
                intent.getParcelable(EXTRA_PARCELIZE) // api 1 - 32

        (
            prop?.a.toString() + prop?.b.toString() + " " + prop?.c + "----\n" +
            serializableProperties?.a + serializableProperties?.b + " " + serializableProperties?.c + "----\n" +
            parcelizeProperties?.a.toString() + parcelizeProperties?.b.toString() + " " + parcelizeProperties?.c
        ).also { binding.textView.text = it }
    }
}